physx-sys/pxbind/main.cpp

```diff
@@ -1119,6 +1119,10 @@ class ClassMatchHandler : public MatchFinder::MatchCallback {
             bool hasSelf = !cmd->isStatic();
             QualType selfType = QualType(cmd->getParent()->getTypeForDecl(), 0);
 
+            const bool needsAllocatingCtor =
+                cmd->getParent()->isPolymorphic() ||
+                !cmd->getParent()->hasSimpleDestructor();
```

Comment:
Does that work? I am not familiar with how physx looks, but this might also need to check each field transitively.

Quickly skimming the api, there is also `bool clang::CXXRecordDecl::isPOD`.

Reply:
It works well enough to generate the current `_new` vs `_new_alloc` functions. IIRC, `isPOD` was way more strict.

Reply:
Okay lets leave it at that. The output looks goods.